src/main/java/com/cen4010/bookstore/book/Book.java

```diff
@@ -3,20 +3,20 @@
 import com.cen4010.bookstore.profileManagement.entity.UserEntity;
 import com.cen4010.bookstore.wishlist.WishList;
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hibernate.annotations.Type;
+
 import java.util.HashSet;
 import java.util.Set;
 import java.util.UUID;
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.ManyToMany;
-import org.hibernate.annotations.Type;
+import javax.persistence.*;
 
 
 @Entity
+@Table
 public class Book {
 
   @Id
+  @GeneratedValue
```

Comment:
What's the purpose of this annotation? For that matter, what's the purpose of making any modifications to the book entity? It's been working fine.

Reply:
So if I don't have @GeneratedValue my HTTP request all come back with 500 response.  The message would say "ids for this class must be manually assigned before calling save()" I found this to be an easy fix for it. I didn't know if this would impact anyone else's code.

Reply:
Were you able to figure out why there was a 500 response? There should be an associated stacktrace.

Reply:
I looked at Axel-Soto's code.  For credit card entity he has @Generated something and for his user entity he's using UUID but he also has a dto for that.